bike-erp/src/components/PermissionManagementPage/UserAccountsTable.tsx

```diff
@@ -0,0 +1,126 @@
+import { Button, Table, TableBody, TableRow } from "@material-ui/core";
+import SaveIcon from "@material-ui/icons/Save";
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { BACKEND_URL } from "../../core/utils/config";
+import PermissionDropdown from "./PermissionDropdown";
+import useStyles from "./UserAccountsTableStyle";
+
+//Table that contains the users' accounts information
+const UserAccountsTable = (props) => {
+  const [data, setData] = useState<any[]>([]);
+  const [initialState] = useState(new Map());
+  const classes = useStyles();
+
+  //fetch the accounts from the database
+  useEffect(() => {
+    axios
+      .get(`${BACKEND_URL}/account_management/admin/accounts`)
+      .then((res) => {
+        setData(res.data.accounts);
+        res.data.accounts.forEach((account: any) => {
+          initialState.set(account.email, account.role);
+        });
+      })
+      .catch((err) => {
+        console.log(err.response.data.message);
+      });
+  }, [initialState]);
+
+  //update the roles in the database
+  const updateRoles = (event) => {
+    event.preventDefault();
+    var table: any = document.getElementById("accounts");
+    /*eslint-disable */
+    for (var i = 1, row: any; (row = table.rows[i]); i++) {
+      let userEmail: String = "";
```

Comment:
not capital for s, for `string`. Capital S is a javascript Object while minuscule is a typescript type